detect vserver before vbox, allowing "vbox-guest" status to overwrite "vserver-host" detection

https://tickets.opscode.com/browse/OHAI-479

--- ruby-ohai-7.0.0.alpha.0/lib/ohai/plugins/linux/virtualization.rb~	2013-11-25 21:34:08.000000000 +0200
+++ ruby-ohai-7.0.0.alpha.0/lib/ohai/plugins/linux/virtualization.rb	2013-11-25 22:36:18.108064611 +0200
@@ -81,6 +81,20 @@
       virtualization[:role] = "guest"
     end
 
+    # Detect Linux-VServer
+    if File.exists?("/proc/self/status")
+      proc_self_status = File.read("/proc/self/status")
+      vxid = proc_self_status.match(/^(s_context|VxID): (\d+)$/)
+      if vxid and vxid[2]
+        virtualization[:system] = "linux-vserver"
+        if vxid[2] == "0"
+          virtualization[:role] = "host"
+        else
+          virtualization[:role] = "guest"
+        end
+      end
+    end
+
     # http://www.dmo.ca/blog/detecting-virtualization-on-linux
     if File.exists?("/usr/sbin/dmidecode")
       so = shell_out("dmidecode")
@@ -109,19 +123,5 @@
         nil
       end
     end
-
-    # Detect Linux-VServer
-    if File.exists?("/proc/self/status")
-      proc_self_status = File.read("/proc/self/status")
-      vxid = proc_self_status.match(/^(s_context|VxID): (\d+)$/)
-      if vxid and vxid[2]
-        virtualization[:system] = "linux-vserver"
-        if vxid[2] == "0"
-          virtualization[:role] = "host"
-        else
-          virtualization[:role] = "guest"
-        end
-      end
-    end
   end
 end
